secure_fw/partitions/internal_trusted_storage/tfm_internal_trusted_storage.c

```diff
@@ -16,12 +16,28 @@
 #include "its_utils.h"
 #include "tfm_sp_log.h"
 
+#ifdef TFM_ITS_ENCRYPTED
+#include "its_crypto_interface.h"
+#endif
+
 #ifdef TFM_PARTITION_PROTECTED_STORAGE
 #include "ps_object_defs.h"
 #endif
 
+#ifdef TFM_ITS_ENCRYPTED
+
+/* Buffer to store the encrypted asset data before it is stored in the
+ * filesystem.
+ */
+static uint8_t enc_asset_data[ITS_UTILS_ALIGN(ITS_BUF_SIZE,
```

Comment:
Do we actually need two buffers? Do we use both of them at the same function call? I see that in get we do but I don't understand why we cannot just use one and skip the memcpy in the end?

Reply:
Needs investigation with Antonio, if in place decryption with `nrf_cc3xx_platform_derived_key_decrypt` and ChaCha20-Poly1305 is supported

Reply:
Ok I changed it to an inline implementation, after we got the ok from arm